src/features/Courses/AddClass/index.jsx

```diff
@@ -33,43 +34,58 @@ const AddClass = ({
+  const [endDate, setEndDate] = useState('');
 
   const handleAddClass = async (e) => {
     e.preventDefault();
     const form = e.target;
-    const dataCreateClass = new FormData(form);
-    const classNameField = dataCreateClass.get('name');
-    const startDate = dataCreateClass.get('start_date');
-    const endDate = dataCreateClass.get('end_date');
-    dataCreateClass.append('course_id', courseInfo.masterCourseId);
-    dataCreateClass.set('start_date', startDate ? new Date(startDate).toISOString() : '');
-    dataCreateClass.set('end_date', endDate ? new Date(endDate).toISOString() : '');
+    const dataClass = new FormData(form);
+    const classNameField = dataClass.get('name');
+    dataClass.set('start_date', startDate ? new Date(startDate).toISOString() : '');
+    dataClass.set('end_date', endDate ? new Date(endDate).toISOString() : '');
 
     if (!classNameField || !startDate) {
       return onClose();
     }
 
-    try {
-      const enrollmentDataInst = new FormData();
-      if (instructorSelected) {
-        enrollmentDataInst.append('unique_student_identifier', instructorSelected.value);
-        enrollmentDataInst.append('rolename', 'staff');
-        enrollmentDataInst.append('action', 'allow');
-      }
-      await dispatch(addClass(dataCreateClass, enrollmentDataInst));
-      onClose();
-      setShowToast(true);
-    } catch (error) {
-      logError(error);
-    } finally {
-      if (isCoursePage) {
-        dispatch(fetchCoursesData(selectedInstitution.id, initialPage));
-        dispatch(updateCoursesCurrentPage(initialPage));
-      } else {
+    if (isEdit) {
+      try {
+        dataClass.append('class_id', courseInfo.classId);
+        await dispatch(editClass(dataClass));
+        onClose();
+        setShowToast(true);
+      } catch (error) {
+        logError(error);
+      } finally {
         dispatch(fetchClassesData(selectedInstitution.id, initialPage, courseId));
         dispatch(updateClassesCurrentPage(initialPage));
       }
+    } else {
+      try {
+        dataClass.append('course_id', courseInfo.masterCourseId);
+        const enrollmentDataInst = new FormData();
+        if (instructorSelected) {
+          enrollmentDataInst.append('unique_student_identifier', instructorSelected.value);
+          enrollmentDataInst.append('rolename', 'staff');
+          enrollmentDataInst.append('action', 'allow');
+        }
+        await dispatch(addClass(dataClass, enrollmentDataInst));
+        onClose();
+        setShowToast(true);
+      } catch (error) {
+        logError(error);
+      } finally {
+        if (isCoursePage) {
+          dispatch(fetchCoursesData(selectedInstitution.id, initialPage));
+          dispatch(updateCoursesCurrentPage(initialPage));
+        } else {
+          dispatch(fetchClassesData(selectedInstitution.id, initialPage, courseId));
+          dispatch(updateClassesCurrentPage(initialPage));
+        }
+      }
     }
+
     return null;
```

Comment:
Just a thought, this could be more clear in this way:

```suggestion
   if (!isEdit) {
      try {
        dataClass.append('course_id', courseInfo.masterCourseId);
        const enrollmentDataInst = new FormData();
        if (instructorSelected) {
          enrollmentDataInst.append('unique_student_identifier', instructorSelected.value);
          enrollmentDataInst.append('rolename', 'staff');
          enrollmentDataInst.append('action', 'allow');
        }
        await dispatch(addClass(dataClass, enrollmentDataInst));
        onClose();
        setShowToast(true);
      } catch (error) {
        logError(error);
      }

      if (!isCoursePage) {
        dispatch(fetchClassesData(selectedInstitution.id, initialPage, courseId));
        return dispatch(updateClassesCurrentPage(initialPage));
      }

      dispatch(fetchCoursesData(selectedInstitution.id, initialPage));
      return dispatch(updateCoursesCurrentPage(initialPage));
    }

    try {
      dataClass.append('class_id', courseInfo.classId);
      await dispatch(editClass(dataClass));
      onClose();
      setShowToast(true);
    } catch (error) {
      logError(error);
    }

    dispatch(fetchClassesData(selectedInstitution.id, initialPage, courseId));
    return dispatch(updateClassesCurrentPage(initialPage));
```

But you can ignore this if you think this is a nonsense